### PR TITLE
Set PackageValidationBaselineName to assemblyName when packageID is not set

### PR DESCRIPTION
**Pull request by Anipik**
Fixes issue in https://github.com/dotnet/runtime/pull/54250

### CODE REVIEW

File: src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <UsingTask TaskName="Microsoft.DotNet.PackageValidation.ValidatePackage" AssemblyFile="$(DotNetPackageValidationAssembly)" />
 
   <PropertyGroup>
+    <PackageValidationBaselineName Condition="'$(PackageId)' == ''">$(AssemblyName)</PackageValidationBaselineName>
     <PackageValidationBaselineName>$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$(PackageId)'))</PackageValidationBaselineName>
```
Comment (ViktorHofer) [inline src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets:7]:
```suggestion
    <PackageValidationBaselineName Condition="'$(PackageId)' == ''">$(AssemblyName)</PackageValidationBaselineName>
    <PackageValidationBaselineName>$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$([MSBuild]::ValueOrDefault('$(PackageId)'))'))</PackageValidationBaselineName>
```

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
In which cases is PackageId not set?

**Comment by Anipik** [comment]
> In which cases is PackageId not set?

Its being set late during the sdk evaluation.

File: src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets [hunk 2]
```diff
@@ -3,7 +3,8 @@
   <UsingTask TaskName="Microsoft.DotNet.PackageValidation.ValidatePackage" AssemblyFile="$(DotNetPackageValidationAssembly)" />
 
   <PropertyGroup>
-    <PackageValidationBaselineName>$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$(PackageId)'))</PackageValidationBaselineName>
+    <PackageValidationBaselineName Condition="'$(PackageId)' == ''">$(AssemblyName)</PackageValidationBaselineName>
```
Comment (ViktorHofer) [inline src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets:6]:
```suggestion
```

**Comment by ViktorHofer** [comment]
> Its being set late during the sdk evaluation.

Doesn't that mean that PackageId is never used then?

**Comment by Anipik** [comment]
> Doesn't that mean that PackageId is never used then?

unless its a global property

**Comment by ViktorHofer** [comment]
> unless its a global property

which is never the case usually.

**Comment by Anipik** [comment]
> which is never the case usually.

setting the packageID in the csproj would be another scenario.

**Comment by ViktorHofer** [comment]
> setting the packageID in the csproj would be another scenario.

That's also very rarely the case for projects unless they would want to override it.

You could easily avoid this sequencing issue by inlining the `PackageValidationBaselineName` value into the two locations where it's used:
- The PackageDownload item
- In the task as a property later

By inlining it, you will only access the `PackageId` value after it is set by NuGet.

File: src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets [hunk 3]
```diff
@@ -2,19 +2,16 @@
...
-    <PackageDownload Include="$(PackageValidationBaselineName)" Version="[$(PackageValidationBaselineVersion)]" />
+    <PackageDownload Include="$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$(PackageId)'))" Version="[$(PackageValidationBaselineVersion)]" />
   </ItemGroup>
 
   <Target Name="RunPackageValidation"
           AfterTargets="Pack"
           Condition="'$(IsPackable)' == 'true'">
 
     <PropertyGroup>
+      <PackageValidationBaselineName>$([MSBuild]::ValueOrDefault('$(PackageValidationBaselineName)', '$(PackageId)'))</PackageValidationBaselineName>
```
Comment (ViktorHofer) [inline src/Compatibility/Microsoft.DotNet.PackageValidation/build/Microsoft.DotNet.PackageValidation.targets:14]:
```suggestion
      <PackageValidationBaselineName Condition="'$(PackageValidationBaselineName)' == ''">$(PackageId)</PackageValidationBaselineName>
```

**Comment by ViktorHofer** [comment]
LGTM module the one remaining feedback item.

**Review by ViktorHofer — APPROVED** [review]

**Review by safern — APPROVED** [review]

**Pull request merged**